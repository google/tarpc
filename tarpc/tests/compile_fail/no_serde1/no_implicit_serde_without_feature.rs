#[tarpc::service]
trait Foo {
    async fn foo();
}

fn main() {
    let x = FooRequest::Foo {};
    x.serialize();
}
