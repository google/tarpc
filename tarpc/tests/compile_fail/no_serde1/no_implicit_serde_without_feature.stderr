error[E0599]: no method named `serialize` found for enum `FooRequest` in the current scope
 --> tests/compile_fail/no_serde1/no_implicit_serde_without_feature.rs:8:7
  |
1 | #[tarpc::service]
  | ----------------- method `serialize` not found for this enum
...
8 |     x.serialize();
  |       ^^^^^^^^^ method not found in `FooRequest`
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following trait defines an item `serialize`, perhaps you need to implement it:
          candidate #1: `serde::ser::Serialize`
